frontend/app/tests/unit/specs/composables/history/filter-paginate-transactions.spec.ts

```diff
@@ -254,5 +254,48 @@ describe('composables::history/filter-paginate', () => {
 
       expect(get(filters).counterparties).toStrictEqual(get(protocols));
     });
+
+    test('exclusion filters', async () => {
+      fetchHistoryEvents = vi.fn();
+
+      const { userAction, fetchData, isLoading, updateFilter } =
+        usePaginationFilters<
+          HistoryEvent,
+          HistoryEventRequestPayload,
+          HistoryEvent,
+          Collection<HistoryEvent>,
+          Filters,
+          Matcher
+        >(
+          locationOverview,
+          mainPage,
+          () => useHistoryEventFilter({ protocols: get(protocols).length > 0 }),
+          fetchHistoryEvents,
+          {
+            onUpdateFilters,
+            extraParams,
+            customPageParams
+          }
+        );
+
+      updateFilter({
+        location: 'protocols',
+        entryTypes: ['!evm event']
+      });
+
+      set(userAction, true);
+      fetchData().catch(() => {});
+      expect(get(isLoading)).toBe(true);
+      await flushPromises();
+      expect(get(isLoading)).toBe(false);
+
+      // @ts-ignore
+      expect(fetchHistoryEvents.mock.calls[0][0].value.entryTypes).toEqual([
```

Comment:
That is the only comment left, otherwise, it looks good. 

Generally, let's try to use the tools at our disposal the way they are designed to be used, and avoid forcing, or ignoring as much as possible. 

If you can't think of a better way to do something let's discuss it and figure out the proper way to do it.

Reply:
I didn't find success with `toHaveBeenCalledWith`. I believe it's because what's sent to the function is a reactive object.

I can think of several solutions to make `toHaveBeenCalledWith` works.
1. mock the spy function (1) implementation to call another spy function (2), but unref the parameter (payload). So I can check the spy function (2) `toHaveBeenCalledWith`
2. https://github.com/rotki/rotki/blob/15fcbe9341031914a93384ed1c52fe1b238d6c44/frontend/app/src/composables/filter-paginate.ts#L192

In this line, do this instead
```
await execute(0, get(pageParams));
```


Anyway I will push with solution 1, unless you tell me to change later 😛

Reply:
You can do partial matching in objects with `toHaveBeenCalledWith` I think I did it in the task tests let me check.

Reply:
Ok, I did but then I replaced it before merging.

But it certainly is possible. I will take care of it no worries.

Reply:
Yeah I did partial matching too (expect.objectContaining), but it doesn't work with reactive object (at least it took long time). But maybe I did it wrong